panic: runtime error: index out of range on starting job

Hi,

I've setup Dkron on three nodes, acting as masters and agents using zookeeper over Centos6. Everything wents ok, but when I schedule a test job, agents crashes with following error:

```
INFO[2016-09-13T14:09:35+01:00] agent: Starting job                           job=apuntador node=node2
panic: runtime error: index out of range

goroutine 43 [running]:
panic(0xbb9dc0, 0xc82000e0c0)
	/Users/victorcoder/src/github.com/mxcl/homebrew/Cellar/go/1.6.3/libexec/src/runtime/panic.go:481 +0x3e6
github.com/victorcoder/dkron/dkron.(*Store).GetExecutions(0xc8201a2960, 0xc8201a0330, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:243 +0x588
github.com/victorcoder/dkron/dkron.(*Store).SetExecution(0xc8201a2960, 0xc820070480, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:329 +0x674
github.com/victorcoder/dkron/dkron.(*AgentCommand).setExecution(0xc820050420, 0xc8200f4370, 0xac, 0xac, 0xc8201a0300)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:602 +0x180
github.com/victorcoder/dkron/dkron.(*AgentCommand).RunQuery(0xc820050420, 0xc820071c80)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/queries.go:92 +0x15e6
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc820056100)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:98 +0x3c0
created by github.com/victorcoder/dkron/cron.(*Cron).run
	/Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x58a
```

Job "apuntador" consists in:

```
#!/bin/bash
#
DATE_T=`date +%c`
echo "It's date "$DATE_T > /tmp/registry.txt
```
Job definition is:

```
{
	"name": "apuntador",
	"schedule": "@every 5s",
	"shell": false,
	"command": "/opt/sds/prueba.sh",
	"owner": "",
	"owner_email": "",
	"success_count": 0,
	"error_count": 0,
	"last_success": "0001-01-01T00:00:00Z",
	"last_error": "0001-01-01T00:00:00Z",
	"disabled": false,
	"tags": {
		"role": "spark-schedule:1"
	},
	"retries": 0,
	"dependent_jobs": null,
	"parent_job": ""
}
```
Which can be the issue?

@victorcoder hey victor, I am encountering the same issue as well. Here is the error I get if it helps.

```bash
./dkron agent -server -backend=zk -backend-machine=127.0.0.1:2181 -log-level=debug
Starting Dkron agent...
INFO[2016-09-20T17:02:48-07:00] agent: Dkron agent starting                   node=MacBook-Pro-2.local
INFO[2016-09-20T17:02:48-07:00] agent: joining: [] replay: true               node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:48-07:00] store: Backend config                         backend=zk keyspace=dcron machines=[127.0.0.1:2181] node=MacBook-Pro-2.local
INFO[2016-09-20T17:02:48-07:00] api: Running HTTP server                      address=:8080 node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:48-07:00] rpc: Registering RPC server                   node=MacBook-Pro-2.local rpc_addr=192.168.1.3:6868
INFO[2016-09-20T17:02:48-07:00] agent: Running for election                   node=MacBook-Pro-2.local
INFO[2016-09-20T17:02:48-07:00] agent: Listen for events                      node=MacBook-Pro-2.local
INFO[2016-09-20T17:02:48-07:00] agent: Cluster leadership lost                node=MacBook-Pro-2.local
INFO[2016-09-20T17:02:48-07:00] agent: Cluster leadership acquired            node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:48-07:00] agent: Restarting scheduler                   node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:48-07:00] scheduler: Adding job to cron                 job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:49-07:00] agent: Received event                         event=member-join node=MacBook-Pro-2.local
DEBU[2016-09-20T17:02:49-07:00] agent: Member event                           event=member-join member=MacBook-Pro-2.local node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] scheduler: Run job                            job=test node=MacBook-Pro-2.local schedule=5 * * * * *
DEBU[2016-09-20T17:03:05-07:00] store: Retrieved job from datastore           job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Filtered nodes to run: []              node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Filtered tags to run: map[]            node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Sending query                          job_name=test json={"execution":{"job_name":"test","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474416185004731313,"attempt":1},"rpc_addr":"192.168.1.3:6868"} node=MacBook-Pro-2.local query=run:job
DEBU[2016-09-20T17:03:05-07:00] agent: Received event                         event=query: run:job node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Running job                            at=1 node=MacBook-Pro-2.local payload={"execution":{"job_name":"test","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474416185004731313,"attempt":1},"rpc_addr":"192.168.1.3:6868"} query=run:job
INFO[2016-09-20T17:03:05-07:00] agent: Starting job                           job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Received ack                           from=MacBook-Pro-2.local node=MacBook-Pro-2.local query=run:job
DEBU[2016-09-20T17:03:05-07:00] rpc: Received GetJob                          job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] store: Retrieved job from datastore           job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: GetJob by RPC                          command=ls node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] agent: Received response                      from=MacBook-Pro-2.local node=MacBook-Pro-2.local query=run:job response={"job_name":"test","started_at":"2016-09-20T17:03:05.007986788-07:00","finished_at":"0001-01-01T00:00:00Z","node_name":"MacBook-Pro-2.local","group":1474416185004731313,"attempt":1}
DEBU[2016-09-20T17:03:05-07:00] store: Setting key                            execution=1474416185007986788-MacBook-Pro-2.local job=test node=MacBook-Pro-2.local
DEBU[2016-09-20T17:03:05-07:00] proc: Command output                          node=MacBook-Pro-2.local output=LICENSE
README.md
config
dkron
static
templates

panic: runtime error: index out of range

goroutine 49 [running]:
panic(0x7bc160, 0xc820010070)
	/Users/victorcoder/src/github.com/mxcl/homebrew/Cellar/go/1.6.3/libexec/src/runtime/panic.go:481 +0x3e6
github.com/victorcoder/dkron/dkron.(*Store).GetExecutions(0xc820017410, 0xc8204dadb0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:243 +0x588
github.com/victorcoder/dkron/dkron.(*Store).SetExecution(0xc820017410, 0xc8201b1280, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:329 +0x674
github.com/victorcoder/dkron/dkron.(*AgentCommand).setExecution(0xc82000a4e0, 0xc820516300, 0xbb, 0xbb, 0xc8204dacf0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:602 +0x180
github.com/victorcoder/dkron/dkron.(*AgentCommand).RunQuery(0xc82000a4e0, 0xc8201b0d00)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/queries.go:92 +0x15e6
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc820506000)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:98 +0x3c0
created by github.com/victorcoder/dkron/cron.(*Cron).run
	/Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x58a
```

I do find the dcron directory inside my zookeeper store. I am wondering if I miss something. 

Thanks a lot.

Hi @victorcoder, here is the result of execution with `-log-level=debug`:

```
Starting Dkron agent...
INFO[2016-09-21T08:19:35+01:00] agent: Dkron agent starting                   node=node3
INFO[2016-09-21T08:19:35+01:00] agent: joining: [172.28.128.3 172.28.128.4 172.28.128.5] replay: true  node=node3
INFO[2016-09-21T08:19:35+01:00] agent: joined: 1 nodes                        node=node3
DEBU[2016-09-21T08:19:35+01:00] store: Backend config                         backend=zk keyspace=dkron machines=[172.28.128.5:2181] node=node3
INFO[2016-09-21T08:19:36+01:00] api: Running HTTP server                      address=172.28.128.5:8090 node=node3
DEBU[2016-09-21T08:19:36+01:00] rpc: Registering RPC server                   node=node3 rpc_addr=172.28.128.5:6868
INFO[2016-09-21T08:19:36+01:00] agent: Running for election                   node=node3
INFO[2016-09-21T08:19:36+01:00] agent: Cluster leadership lost                node=node3
INFO[2016-09-21T08:19:36+01:00] agent: Listen for events                      node=node3
INFO[2016-09-21T08:19:36+01:00] agent: Cluster leadership acquired            node=node3
DEBU[2016-09-21T08:19:36+01:00] agent: Restarting scheduler                   node=node3
DEBU[2016-09-21T08:19:36+01:00] store: No jobs found                          node=node3
DEBU[2016-09-21T08:19:36+01:00] agent: Received event                         event=member-join node=node3
DEBU[2016-09-21T08:19:36+01:00] agent: Member event                           event=member-join member=node2 node=node3
DEBU[2016-09-21T08:19:36+01:00] agent: Member event                           event=member-join member=node1 node=node3
DEBU[2016-09-21T08:19:36+01:00] agent: Member event                           event=member-join member=node3 node=node3
DEBU[2016-09-21T08:24:57+01:00] agent: Received event                         event=query: scheduler:restart node=node3
DEBU[2016-09-21T08:24:57+01:00] agent: Restarting scheduler                   node=node3
DEBU[2016-09-21T08:24:57+01:00] scheduler: Stopping scheduler                 node=node3
DEBU[2016-09-21T08:24:57+01:00] scheduler: Adding job to cron                 job=apuntador node=node3
DEBU[2016-09-21T08:25:02+01:00] scheduler: Run job                            job=apuntador node=node3 schedule=@every 5s
DEBU[2016-09-21T08:25:02+01:00] store: Retrieved job from datastore           job=apuntador node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Filtered nodes to run: []              node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Filtered tags to run: map[role:spark-schedule:1]  node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Sending query                          job_name=apuntador json={"execution":{"job_name":"apuntador","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474442702000218752,"attempt":1},"rpc_addr":"172.28.128.5:6868"} node=node3 query=run:job
DEBU[2016-09-21T08:25:02+01:00] agent: Received event                         event=query: run:job node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Running job                            at=2 node=node3 payload={"execution":{"job_name":"apuntador","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474442702000218752,"attempt":1},"rpc_addr":"172.28.128.5:6868"} query=run:job
INFO[2016-09-21T08:25:02+01:00] agent: Starting job                           job=apuntador node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Received ack                           from=node3 node=node3 query=run:job
DEBU[2016-09-21T08:25:02+01:00] rpc: Received GetJob                          job=apuntador node=node3
DEBU[2016-09-21T08:25:02+01:00] store: Retrieved job from datastore           job=apuntador node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: GetJob by RPC                          command=/opt/sds/prueba.sh node=node3
DEBU[2016-09-21T08:25:02+01:00] agent: Received response                      from=node3 node=node3 query=run:job response={"job_name":"apuntador","started_at":"2016-09-21T08:25:02.002865825+01:00","finished_at":"0001-01-01T00:00:00Z","node_name":"node3","group":1474442702000218752,"attempt":1}
DEBU[2016-09-21T08:25:02+01:00] store: Setting key                            execution=1474442702002865825-node3 job=apuntador node=node3
panic: runtime error: index out of range

goroutine 41 [running]:
panic(0xbb9dc0, 0xc8200100d0)
	/Users/victorcoder/src/github.com/mxcl/homebrew/Cellar/go/1.6.3/libexec/src/runtime/panic.go:481 +0x3e6
github.com/victorcoder/dkron/dkron.(*Store).GetExecutions(0xc8201cc000, 0xc82025faa0, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:243 +0x588
github.com/victorcoder/dkron/dkron.(*Store).SetExecution(0xc8201cc000, 0xc820071080, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:329 +0x674
github.com/victorcoder/dkron/dkron.(*AgentCommand).setExecution(0xc82014e1e0, 0xc8200e3ce0, 0xac, 0xac, 0xc82025f9e0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:602 +0x180
github.com/victorcoder/dkron/dkron.(*AgentCommand).RunQuery(0xc82014e1e0, 0xc82013cb00)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/queries.go:92 +0x15e6
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc82013a100)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:98 +0x3c0
created by github.com/victorcoder/dkron/cron.(*Cron).run
	/Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x58a
```
Prior to the execution, I've removed dkron entry in zookeeper so a clean installation is done. After execution, Zookeeper entrey is created, and a master is elected. It starts crashing when I submit a new job, when a host tries to run it, it crashes. When the crasehd node is a master, another master is elected correctly, so I think is not a zookeeper connectivity issue.

Thank you!

@victorcoder @MingChenSlot Using only one node, against itself as zookeeper backend, still same issue:
```
Starting Dkron agent...
INFO[2016-09-21T09:06:27+01:00] agent: Dkron agent starting                   node=node1
INFO[2016-09-21T09:06:27+01:00] agent: joining: [172.28.128.3] replay: true   node=node1
INFO[2016-09-21T09:06:27+01:00] agent: joined: 1 nodes                        node=node1
DEBU[2016-09-21T09:06:27+01:00] store: Backend config                         backend=zk keyspace=dkron machines=[172.28.128.3:2181] node=node1
INFO[2016-09-21T09:06:27+01:00] api: Running HTTP server                      address=172.28.128.3:8090 node=node1
DEBU[2016-09-21T09:06:27+01:00] rpc: Registering RPC server                   node=node1 rpc_addr=172.28.128.3:6868
INFO[2016-09-21T09:06:27+01:00] agent: Running for election                   node=node1
INFO[2016-09-21T09:06:27+01:00] agent: Listen for events                      node=node1
INFO[2016-09-21T09:06:27+01:00] agent: Cluster leadership lost                node=node1
INFO[2016-09-21T09:06:27+01:00] agent: Cluster leadership acquired            node=node1
DEBU[2016-09-21T09:06:27+01:00] agent: Restarting scheduler                   node=node1
DEBU[2016-09-21T09:06:27+01:00] scheduler: Adding job to cron                 job=apuntador node=node1
DEBU[2016-09-21T09:06:28+01:00] agent: Received event                         event=member-join node=node1
DEBU[2016-09-21T09:06:28+01:00] agent: Member event                           event=member-join member=node1 node=node1
DEBU[2016-09-21T09:06:32+01:00] scheduler: Run job                            job=apuntador node=node1 schedule=@every 5s
DEBU[2016-09-21T09:06:32+01:00] store: Retrieved job from datastore           job=apuntador node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Filtered nodes to run: []              node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Filtered tags to run: map[role:spark-schedule:1]  node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Sending query                          job_name=apuntador json={"execution":{"job_name":"apuntador","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474445192000288087,"attempt":1},"rpc_addr":"172.28.128.3:6868"} node=node1 query=run:job
DEBU[2016-09-21T09:06:32+01:00] agent: Received event                         event=query: run:job node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Running job                            at=1 node=node1 payload={"execution":{"job_name":"apuntador","started_at":"0001-01-01T00:00:00Z","finished_at":"0001-01-01T00:00:00Z","group":1474445192000288087,"attempt":1},"rpc_addr":"172.28.128.3:6868"} query=run:job
INFO[2016-09-21T09:06:32+01:00] agent: Starting job                           job=apuntador node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Received ack                           from=node1 node=node1 query=run:job
DEBU[2016-09-21T09:06:32+01:00] rpc: Received GetJob                          job=apuntador node=node1
DEBU[2016-09-21T09:06:32+01:00] store: Retrieved job from datastore           job=apuntador node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: GetJob by RPC                          command=/opt/sds/prueba.sh node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: Received response                      from=node1 node=node1 query=run:job response={"job_name":"apuntador","started_at":"2016-09-21T09:06:32.01700035+01:00","finished_at":"0001-01-01T00:00:00Z","node_name":"node1","group":1474445192000288087,"attempt":1}
DEBU[2016-09-21T09:06:32+01:00] store: Setting key                            execution=1474445192017000350-node1 job=apuntador node=node1
DEBU[2016-09-21T09:06:32+01:00] proc: Command output                          node=node1 output=
DEBU[2016-09-21T09:06:32+01:00] agent: Received event                         event=query: rpc:config node=node1
DEBU[2016-09-21T09:06:32+01:00] agent: RPC Config requested                   at=2 node=node1 payload= query=rpc:config
DEBU[2016-09-21T09:06:32+01:00] proc: Received ack                            from=node1 node=node1 query=rpc:config
DEBU[2016-09-21T09:06:32+01:00] proc: Received response                       from=node1 node=node1 payload=172.28.128.3:6868 query=rpc:config
panic: runtime error: index out of range

goroutine 50 [running]:
panic(0xbb9dc0, 0xc8200100c0)
	/Users/victorcoder/src/github.com/mxcl/homebrew/Cellar/go/1.6.3/libexec/src/runtime/panic.go:481 +0x3e6
github.com/victorcoder/dkron/dkron.(*Store).GetExecutions(0xc82011a8d0, 0xc820175220, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:243 +0x588
github.com/victorcoder/dkron/dkron.(*Store).SetExecution(0xc82011a8d0, 0xc820072600, 0x0, 0x0, 0x0, 0x0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:329 +0x674
github.com/victorcoder/dkron/dkron.(*AgentCommand).setExecution(0xc82013e1e0, 0xc8200b6580, 0xab, 0xab, 0xc820175160)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:602 +0x180
github.com/victorcoder/dkron/dkron.(*AgentCommand).RunQuery(0xc82013e1e0, 0xc82012af00)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/queries.go:92 +0x15e6
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc82012d600)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:98 +0x3c0
created by github.com/victorcoder/dkron/cron.(*Cron).run
	/Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x58a
```
Regards

--- a/config/dkron.json
+++ b/config/dkron.json
@@ -3,6 +3,6 @@
     "role": "web",
     "datacenter": "east"
   },
-  "keyspace": "dcron",
+  "keyspace": "dkron",
   "encrypt": "kPpdjphiipNSsjd4QHWbkA=="
 }

--- a/dkron/store.go
+++ b/dkron/store.go
@@ -21,6 +21,7 @@ type Store struct {
 	Client   store.Store
 	agent    *AgentCommand
 	keyspace string
+	backend  string
 }
 
 func init() {
@@ -46,7 +47,7 @@ func NewStore(backend string, machines []string, a *AgentCommand, keyspace strin
 		log.WithError(err).Fatal("store: Store backend not reachable")
 	}
 
-	return &Store{Client: s, agent: a, keyspace: keyspace}
+	return &Store{Client: s, agent: a, keyspace: keyspace, backend: backend}
 }
 
 // Store a job
@@ -186,7 +187,7 @@ func (s *Store) GetJobs() ([]*Job, error) {
 		return nil, err
 	}
 
-	var jobs []*Job
+	jobs := make([]*Job, 0)
 	for _, node := range res {
 		var job Job
 		err := json.Unmarshal([]byte(node.Value), &job)
@@ -248,10 +249,12 @@ func (s *Store) GetExecutions(jobName string) ([]*Execution, error) {
 	var executions []*Execution
 
 	for _, node := range res {
-		path := store.SplitKey(node.Key)
-		dir := path[len(path)-2]
-		if dir != jobName {
-			continue
+		if store.Backend(s.backend) != store.ZK {
+			path := store.SplitKey(node.Key)
+			dir := path[len(path)-2]
+			if dir != jobName {
+				continue
+			}
 		}
 		var execution Execution
 		err := json.Unmarshal([]byte(node.Value), &execution)

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -8,7 +8,7 @@ services:
       - "4001"
     volumes:
       - ./etcd.data:/data
-    command: -name=dcron1
+    command: -name=dkron1
 
   consul:
     image: consul
@@ -18,11 +18,20 @@ services:
     hostname: node1
     command: agent -server -bootstrap -ui-dir=/ui -client=0.0.0.0
 
+  zk:
+    image: zookeeper
+    ports:
+      - "2181"
+    hostname: node1
+
   dkron:
     depends_on:
-      - consul
+      # Uncomment to use consul
+      #- consul
       # Uncommend to use etcd
       #- etcd
+      # Uncommend to use zk
+      - zk
     ports:
       - "8080"
     build: .
@@ -30,6 +39,9 @@ services:
      - ./:/gopath/src/github.com/victorcoder/dkron
     environment:
       - GODEBUG=netdns=go
+    # Uncomment to use consul
     command: bash -c "go build *.go && ./main agent -server -backend=consul -backend-machine=consul:8500 -join=dkron:8946 -log-level=debug"
     # Uncomment to use etcd
-    #command: bash -c "go build *.go && ./main agent -server -backend=etcd -backend-machine=etcd:4001 -join=dkron_seed:8946 -log-level=debug"
+    # command: bash -c "go build *.go && ./main agent -server -backend=etcd -backend-machine=etcd:4001 -join=dkron:8946 -log-level=debug"
+    # Uncomment to use zk
+    # command: bash -c "go build *.go && ./main agent -server -backend=zk -backend-machine=zk:2181 -join=dkron:8946 -log-level=debug"

